k8sclient/k8sclient.go

```diff
@@ -178,18 +176,34 @@ func getCNIConfigFromFile(name string, confdir string) ([]byte, error) {
 	}
 
 	for _, confFile := range files {
-		conf, err := libcni.ConfFromFile(confFile)
-		if err != nil {
-			return nil, fmt.Errorf("Error loading CNI config file %s: %v", confFile, err)
-		}
+		var confList *libcni.NetworkConfigList
+		if strings.HasSuffix(confFile, ".conflist") {
+			confList, err = libcni.ConfListFromFile(confFile)
+			if err != nil {
+				return nil, fmt.Errorf("Error loading CNI conflist file %s: %v", confFile, err)
+			}
+
+			if confList.Name == name {
+				if len(confList.Plugins) == 0 {
+					return nil, fmt.Errorf("CNI config list %s has no networks, skipping", confFile)
+				}
+				return confList.Bytes, nil
+			}
 
-		if conf.Network.Name == name {
-			// Ensure the config has a "type" so we know what plugin to run.
-			// Also catches the case where somebody put a conflist into a conf file.
-			if conf.Network.Type == "" {
-				return nil, fmt.Errorf("Error loading CNI config file %s: no 'type'; perhaps this is a .conflist?", confFile)
+		} else {
+			conf, err := libcni.ConfFromFile(confFile)
+			if err != nil {
+				return nil, fmt.Errorf("Error loading CNI config file %s: %v", confFile, err)
+			}
+
+			if conf.Network.Name == name {
+				// Ensure the config has a "type" so we know what plugin to run.
+				// Also catches the case where somebody put a conflist into a conf file.
+				if conf.Network.Type == "" {
+					return nil, fmt.Errorf("Error loading CNI config file %s: no 'type'; perhaps this is a .conflist?", confFile)
+				}
+				return conf.Bytes, nil
```

Comment:
We can actually use ConfLists everywhere  by "upconverting" the single conf with `libcni.ConfListFromConf(conf)`

Reply:
@dcbw  sure.. I will check the possibility..